localstack/services/sqs/query_api.py

```diff
@@ -0,0 +1,142 @@
+"""The SQS Query API allows using Queue URLs as endpoints for operations on that queue. See:
+https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-making-api-requests.html. This is a
+generic implementation that creates from Query API requests the respective AWS requests, and uses an aws_stack client
+to make the request. """
+import logging
+from typing import Dict, Tuple
+from urllib.parse import urlencode
+
+from botocore.exceptions import ClientError
+from botocore.model import OperationModel
+
+from localstack import config
+from localstack.aws.api import CommonServiceException
+from localstack.aws.protocol.parser import OperationNotFoundParserError, create_parser
+from localstack.aws.protocol.serializer import create_serializer
+from localstack.aws.spec import load_service
+from localstack.http import Request, Response, Router, route
+from localstack.http.dispatcher import Handler
+from localstack.utils.aws import aws_stack
+
+LOG = logging.getLogger(__name__)
+
+service = load_service("sqs")
+parser = create_parser(service)
+serializer = create_serializer(service)
+
+
+@route(
+    '/queue/<regex("[a-z0-9-]+"):region>/<regex("[0-9]{12}"):account_id>/<regex("[a-zA-Z0-9_-]+(.fifo)?"):queue_name>',
+    methods=["POST", "GET"],
+)
+def path_strategy_handler(request: Request, region, account_id, queue_name):
+    return handle_request(request, region)
+
+
+@route(
+    '/<regex("[0-9]{12}"):account_id>/<regex("[a-zA-Z0-9_-]+(.fifo)?"):queue_name>',
+    host="sqs.<regex('([a-z0-9-]+)?'):region>.localstack.cloud<regex('(:[0-9]{2,5})?'):port>",
+    methods=["POST", "GET"],
+)
+def domain_strategy_handler(request: Request, account_id, queue_name, region=None, port=None):
+    """Uses the endpoint host to extract the region. See:
+    https://docs.aws.amazon.com/general/latest/gr/sqs-service.html"""
+    return handle_request(request, region)
+
+
+@route(
+    '/<regex("[0-9]{12}"):account_id>/<regex("[a-zA-Z0-9_-]+(.fifo)?"):queue_name>',
+    methods=["POST", "GET"],
+)
+def legacy_handler(request: Request, account_id, queue_name) -> Response:
```

Comment:
The legacy handler has return type -> Response, the domain and path one do not. Just checking if that is intentional

Reply:
lazyness! will fix